main.tf

```diff
@@ -26,6 +26,8 @@ locals {
   ) : null
   # tflint-ignore: terraform_unused_declarations
   validate_metrics = var.plan != "enterprise-3nodes-2tb" && length(var.metrics) > 0 ? tobool("metrics are only supported for enterprise plan") : true
+  # tflint-ignore: terraform_unused_declarations
+  validate_schema_global_rule = var.plan != "enterprise-3nodes-2tb" && var.schema_global_rule != "" ? tobool("schema global rule is only supported for enterprise plan") : true
```

Comment:
as per https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/pull/342/files#r1848885092 - default to `null` and not `""`